testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java

```diff
@@ -125,12 +126,9 @@ public List<List<SpanData>> waitForTraces(int numberOfTraces) {
    * This waits up to 20 seconds, then times out.
    */
   public List<LogRecordData> waitForLogRecords(int numberOfLogRecords) {
```

Comment:
Firstly I think you should update the PR title or description to make it easier to understand what you are doing. As far as I can tell the improvement is that you are making metric and log assertion similar to trace assertion.
WDYT about moving the log assertion logic to `InstrumentationTestRunner` as it is for traces/metrics. Then you would not need the `AwaitUtil` class and could have the helper method in `InstrumentationTestRunner`.

Reply:
good point @laurit - fixed